src/resmom/linux/alps.c

```diff
@@ -4208,13 +4277,20 @@ alps_create_reserve_request(job *pjob, basil_request_reserve_t **req)
 
 		/*
 		 * Only match vnodes that begin with mpphost and have
-		 * a following "_<num>_<num>".
+		 * a following "_<num>_<num>" (when vnode_per_numa_node
+		 * is true) otherwise, just plain "_<num>".
```

Comment:
For this comment, I would suggest: Skip over vnodes where the name does not begin with the expected mpphost string.
Then add another comment as indicated below.